src/ralph/backends/http/async_lrs.py

```diff
@@ -157,14 +157,14 @@ async def read(  # pylint: disable=too-many-arguments
         if not target:
             target = self.settings.STATEMENTS_ENDPOINT
 
-        if query and query.limit:
+        if query.limit:
```

Comment:
What if `query` is not provided (at `None`)?

Reply:
I think the magic `@enforce_query_checks` decorator should handle this case. It calls `validate_query` before passing the `query` to this method) We intend to change that with the template pattern in the future)

Reply:
Oh true, missed it again!